src/main/java/com/easypost/http/Requestor.java

```diff
@@ -428,18 +448,25 @@ protected static <T> T request(final RequestMethod method, final String url,
         }
     }
 
+    /**
+     * Send an HTTP request to EasyPost.
+     *
+     * @param <T>    Any class.
+     * @param method The method of the API request.
+     * @param url    The URL of the API request.
+     * @param params The params of the API request.
+     * @param clazz  The class of the object for deserialization
+     * @param client The EasyPostClient object.
+     *
+     * @return A class object.
+     * @throws EasyPostException when the request fails.
+     */
     @SuppressWarnings("checkstyle:methodname")
-    protected static <T> T _request(final RequestMethod method, final String url,
-            final Map<String, Object> params, final Class<T> clazz, String apiKey,
-            final boolean apiKeyRequired) throws EasyPostException {
-        if ((EasyPost.apiKey == null || EasyPost.apiKey.length() == 0) && (apiKey == null || apiKey.length() == 0)) {
-            if (apiKeyRequired) {
-                throw new MissingParameterError(Constants.INVALID_API_KEY_TYPE);
-            }
-        }
-
-        if (apiKey == null) {
-            apiKey = EasyPost.apiKey;
+    protected static <T> T httpRequest(final RequestMethod method, final String url,
+            final Map<String, Object> params, final Class<T> clazz, final EasyPostClient client)
+            throws EasyPostException {
+        if (client.getApiKey() == null || client.getApiKey().isEmpty()) {
+            throw new MissingParameterError(Constants.INVALID_API_KEY_TYPE);
```

Comment:
This check should happen higher up the stack - you can't create a client object without an API key, so this check shouldn't be necessary and can be removed (it's a duplicate).

Reply:
Users could initialize a EasyPostClient object with a `null` or `empty` string, this check statement is needed in case a user pass in an invalid `API Key` when creating EasyPostClient

Reply:
The null check should happen (and throw an error) during Client construction.

Reply:
Yea, we could. But the user will need to add an exception handler in the EasyPostClient initialization method. Any thought?

Reply:
API key is required, so don't let users create a Client without one (don't provide that overload)

Reply:
If it throws an error because the user did something incorrectly, it's on the user to account for doing things incorrectly, you know?